### PR TITLE
Bump torch-mlir to 98e08023bbf71e00ab81e980eac9f7c96f1f24b4

### PR DESCRIPTION
**Pull request by jinchen62**

### CODE REVIEW

**Review by MaheshRavishankar — APPROVED** [review]

File: third_party/torch-mlir
Comment (ScottTodd) [inline third_party/torch-mlir:1]: https://github.com/llvm/torch-mlir/commit/fa39d91357e8ebbf375f211274dfb4bbbbbc5ccf included here seems to have broken some downstream usage of the IREE Python bindings: https://github.com/iree-org/iree-turbine/actions/runs/10617974632/job/29432221240?pr=112#step:5:204

We may be missing some test coverage in this repo.

Can someone verify / repro, then decide if we should revert or fix forward?

**Comment by MaheshRavishankar** [comment]
I'd say we revert this change.

**Comment by jinchen62** [comment]
@MaheshRavishankar @ScottTodd So should I revert this PR or revert that torch-mlir commit?

File: third_party/torch-mlir
Comment (ScottTodd) [reply third_party/torch-mlir:1]: I can repro with just IREE:

1. Build from source: https://iree.dev/building-from-source/getting-started/#building-with-cmake
2. Setup venv and install some deps:
    ```bash
    python -m venv .venv
    .venv\Scripts\activate.bat
    D:\dev\projects\iree-build\.env.bat
    python -m pip install onnx torch sympy
    ```
3. Try running
    ```python
    >>> from iree.compiler.extras.fx_importer import ContextCache
    
    Traceback (most recent call last):  File "<stdin>", line 1, in <module>
      File "D:\dev\projects\iree-build\compiler\bindings\python\iree\compiler\extras\fx_importer.py", line 38, in <module>
        import torch
      File "D:\dev\scratch\iree\.venv\Lib\site-packages\torch\__init__.py", line 2120, in <module>
        from torch._higher_order_ops import cond
      File "D:\dev\scratch\iree\.venv\Lib\site-packages\torch\_higher_order_ops\__init__.py", line 1, in <module>
        from .cond import cond
      File "D:\dev\scratch\iree\.venv\Lib\site-packages\torch\_higher_order_ops\cond.py", line 5, in <module>
        import torch._subclasses.functional_tensor
      File "D:\dev\scratch\iree\.venv\Lib\site-packages\torch\_subclasses\functional_tensor.py", line 42, in <module>
        class FunctionalTensor(torch.Tensor):
      File "D:\dev\scratch\iree\.venv\Lib\site-packages\torch\_subclasses\functional_tensor.py", line 258, in FunctionalTensor
        cpu = _conversion_method_template(device=torch.device("cpu"))
    D:\dev\scratch\iree\.venv\Lib\site-packages\torch\_subclasses\functional_tensor.py:258: UserWarning: Failed to initialize NumPy: _ARRAY_API not found (Triggered internally at C:\actions-runner\_work\pytorch\pytorch\builder\windows\pytorch\torch\csrc\utils\tensor_numpy.cpp:84.)
      cpu = _conversion_method_template(device=torch.device("cpu"))
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "D:\dev\projects\iree-build\compiler\bindings\python\iree\compiler\extras\fx_importer.py", line 138, in <module>
        from .._mlir_libs._torchMlir import get_int64_max, get_int64_min
    ModuleNotFoundError: No module named 'iree.compiler._mlir_libs._torchMlir'
    ```

Comment (ScottTodd) [reply third_party/torch-mlir:1]: I can't repro from running https://github.com/iree-org/iree/blob/main/compiler/bindings/python/test/extras/fx_importer_test.py though.

**Comment by ScottTodd** [comment]
> @MaheshRavishankar @ScottTodd So should I revert this PR or revert that torch-mlir commit?

Added a comment to the upstream PR. Still seeing if we can work around it locally in the meantime.

**Comment by ScottTodd** [comment]
But while we wait on movement upstream (or contribute a patch ourselves), I vote we revert this since it breaks some usage of IREE's Python APIs that iree-turbine depends on.

File: third_party/torch-mlir
Comment (ScottTodd) [reply third_party/torch-mlir:1]: Ah. https://github.com/iree-org/iree/blob/main/compiler/bindings/python/test/extras/fx_importer_test.py _does_ catch this, _if `torch` is installed_. Should get that test really running, together with https://github.com/iree-org/iree/issues/16624. Maybe a new workflow similar to https://github.com/iree-org/iree/blob/main/.github/workflows/pkgci_test_tensorflow_cpu.yml focused on torch.

Comment (ScottTodd) [reply third_party/torch-mlir:1]: Adding / enabling test coverage that would have caught this in https://github.com/iree-org/iree/pull/18420

**Pull request merged**